node/src/components/linear_chain_sync.rs

```diff
@@ -510,6 +587,7 @@ where
                         // If we do, it's a bug.
                         assert_eq!(*block.hash(), block_hash, "Block hash mismatch.");
                         trace!(%block_hash, "linear block found in the local storage.");
+                        // TODO if block is a switch block, start downloading descendants.
```

Comment:
Ah, that might be a leftover of the stuff we were doing with Mateusz.

Reply:
I think what you put under `BlockByHeight` should be in here :grimacing:

Reply:
Actually, not really - in fact, we shouldn't be doing what I did there at all. This is handled by the fact that the Block Executor doesn't try to re-execute blocks that are already in storage. We do have to go back to the last switch block, though, in order to get the correct validator weights (although that might be wrong when an upgrade modifies the global state :weary: but this will have to wait for the change to the auction contract state).